prez-lib/src/types.ts

```diff
@@ -1,183 +1,87 @@
+ */
+export type PrezNode = PrezTerm & {
+    iri: string;
     label?: PrezLiteral;
     description?: PrezLiteral;
-    uri: string;
+    provenance?: PrezLiteral;
     curie?: string;
-}
-
-export interface PrezList {
-    class: PrezTerm;
-    title?: PrezLiteral;
-    description?: PrezLiteral;
-    rows: PrezListRow[];
-    headers: Record<string, PrezNode>;
-}
-
-export interface PrezListRow {
-    [key: string]: PrezTerm;
-}
-
-export interface PrezObject {
-    class: PrezTerm;
-    title: PrezLiteral;
-    description?: PrezLiteral;
-    properties: PrezProperties;
-}
-
-// index each property based on their name/curie for simple access
-export type PrezProperties = Record<string, PrezProperty>;
-
-export interface PrezProperty {
-    predicate: PrezNode;
-    object: PrezTerm|PrezTerm[];
-}
-
-export interface PrezBlankNode extends PrezTerm {
-    termType: 'BlankNode';
-    properties: PrezProperties;
-}
-
-export const PrezDataFactory = {
-    prezNode(paramsOrURI:{uri: string, label?: PrezLiteral|string, description?: PrezLiteral|string, curie?: string}|string): PrezNode {
-        if(typeof(paramsOrURI) == 'string') {
-            return {
-                termType: 'Node',
-                uri: paramsOrURI
-            }
-        }
-        const {uri, label, description, curie} = paramsOrURI;
-        return {
-          termType: 'Node',
-          uri,
-          curie,
-          label: (typeof(label) == 'string' ? this.prezLiteral(label) : label),
-          description: (typeof(description) == 'string' ? this.prezLiteral(description) : description),
-        };
-    },
-
-    prezLiteral(paramsOrString:{text: string, language?: string, dataType?: PrezNode}|string): PrezLiteral {
-        if(typeof(paramsOrString) == 'string') {
-            return {
-                termType: 'Literal',
-                text: paramsOrString
-            }
-        }
-        const {text, language, dataType} = paramsOrString;
-        return {
-            termType: 'Literal',
-            text,
-            language,
-            dataType
-        };
-    },
-
-    prezBlankNode(node: PrezProperties): PrezBlankNode {
-        return {
-          termType: 'BlankNode',
-          properties
-        };
-    }
-
+    links?: PrezLink[];
+    rdfTypes?: PrezNode[];
+    termType: "node";
 };
 
-
-
-
-
-export interface ListItem {
-    label?: string;
-    uri: string;
-    description?: string;
-    link?: string;
-    extras?: {
-        [key: string]: string;
-    };
+/**
+ * Represents a `prez:link` value, containing parent info
+ */
+export type PrezLink = {
+    value: string;
+    parents?: PrezNode[];
+    // maybe label?
 };
 
-export interface ObjectItem {
-    label?: string;
-    uri: string;
-    description?: string;
-    link?: string;
-    properties: {
-        predicate: string;
-        object: string;
-    }[];
+/**
+ * Represents an RDF Blank Node
+ */
+export type PrezBlankNode = PrezTerm & {
+    id: string;
```

Comment:
Does each property have a unique predicate name? Previously I defined this structure for properties so you could quickly access a property using the predicate...

// index each property based on their name/curie for simple access
export type PrezProperties = Record<string, PrezProperty>;

Reply:
as a curie or URI they will be unique yes (obviously the predicate can have multiple values ("objects") - don't think that's an issue here)